lib/Rex/CLI.pm

```diff
@@ -678,10 +678,49 @@ sub load_rexfile {
   }
 
   # load Rexfile
-  eval { require $rexfile };
+  eval {
+    unshift @INC, sub {
+      my $load_file = $_[1];
+      if ( $load_file eq $rexfile ) {
+        open( my $fh, "<", $rexfile ) or die("Error can't open $rexfile: $!");
+        my @content = <$fh>;
+        close($fh);
+        chomp @content;
+
+        push @content, "42;"; # add a true return value at the end of $rexfile.
+          # we need to do this because perl want a "true" value
+          # at the end of a file that is loaded.
+
+        shift @INC; # remove this loader from @INC
+
+        return \join( "\n", @content );
```

Comment:
this is on purpose, this creates a scalar reference which the perl module loader accepts as something like a file handle.